test/test_local_io.py

```diff
@@ -655,6 +655,42 @@ def filepath_fn(name: str) -> str:
         # __len__ Test: returns the length of source DataPipe
         self.assertEqual(3, len(saver_dp))
 
+    @skipIfNoIoPath
+    def test_io_path_saver_file_lock(self):
+        def filepath_fn(name: str) -> str:
+            return os.path.join(self.temp_dir.name, os.path.basename(name))
```

Comment:
Could you please also move this function out of local scope? Python's multiprocessing doesn't support well with local functions.